horovod/mxnet/mpi_ops.cc

```diff
@@ -54,55 +60,56 @@ void DoAllreduce(NDArray* tensor, NDArray* output, const std::string& name,
   auto enqueue_result =
       EnqueueTensorAllreduce(hvd_context, hvd_tensor, hvd_output, nullptr,
                              name, device,
-                             [cb](const Status& status) {
-                               cb();
+                             [on_complete](const Status& status) {
+                               InvokeCompleteCallback(on_complete, status);
                              });
   ThrowIfError(enqueue_result);
 }
 
 #if HAVE_CUDA
 void DoAllreduceCudaOnCPU(NDArray* tensor, NDArray* output, std::string& name,
-                          Callback cb) {
+                          Callback on_complete) {
   ThrowIfError(common::CheckInitialized());
+
   // Make async copy of input tensor to CPU tensor and record completion event.
   auto hvd_cpu_buffer = std::make_shared<MXTemporaryBuffer<NDArray>>(
       CPU_DEVICE_ID, tensor->dtype());
   TensorUtil::AsyncCopyCudaToCPU(tensor, hvd_cpu_buffer->tensor());
-  auto ready_event = std::make_shared<MXReadyEvent<NDArray>>(tensor);
-
   auto hvd_context = std::make_shared<MXOpContext<NDArray>>(
       CPU_DEVICE_ID, hvd_cpu_buffer->tensor());
+  auto ready_event = std::make_shared<MXReadyEvent<NDArray>>(tensor);
```

Comment:
Does the order or this line make difference?

Reply:
nope. just cosmetic change and followed the order in another function.